mmv1/templates/terraform/constants/spanner_database.go.erb

```diff
@@ -80,4 +80,19 @@ func validateDatabaseRetentionPeriod(v interface{}, k string) (ws []string, erro
 	}
 
 	return
+}
+
+func resourceSpannerDBVirtualUpdate(d *schema.ResourceData, resourceSchema map[string]*schema.Schema) bool {
+	// deletion_protection is the only virtual field
+	if d.HasChange("deletion_protection") {
```

Comment:
I think the logic here may not be correct. If I'm reading this right, the logic is:

- if deletion_protection is changed, then loop through all the fields in the schema and always return `true`
- otherwise (if deletion_protection isn't changed) always return false

If the goal is to detect when deletion_protection is the only field being changed - I think you might need to do something like:

```go
if d.HasChange("deletion_protection") {
  for field := range resourceSchema {
    if field == "deletion_protection" {
      continue
    }
    if d.HasChange(field) {
      return false
    }
  }
  return true
}
```

Reply:
updated